src/aks-preview/azext_aks_preview/decorator.py

```diff
@@ -1301,6 +1351,60 @@ def set_up_network_profile(self, mc: ManagedCluster) -> ManagedCluster:
         mc = super().set_up_network_profile(mc)
         network_profile = mc.network_profile
 
+        (
+            pod_cidr,
+            service_cidr,
+            dns_service_ip,
+            _,
+            _,
+        ) = self.context._get_pod_cidr_and_service_cidr_and_dns_service_ip_and_docker_bridge_address_and_network_policy(enable_validation=False)
+
+        (
+            pod_cidrs,
+            service_cidrs,
+            ip_families
+        ) = self.context.get_pod_cidrs_and_service_cidrs_and_ip_families()
+
+        # set dns_service_ip, pod_cidr(s), service(s) with user provided values if
+        # of them are set. Largely follows the base function which will potentially
+        # overwrite default SDK values.
+        if any([
+            dns_service_ip,
+            pod_cidr,
+            pod_cidrs,
+            service_cidr,
+            service_cidrs,
+        ]):
+            network_profile.dns_service_ip = dns_service_ip
+            network_profile.pod_cidr = pod_cidr
+            network_profile.pod_cidrs = pod_cidrs
+            network_profile.service_cidr = service_cidr
+            network_profile.service_cidrs = service_cidrs
```

Comment:
I remember that in the previous implementation, the value of `pod_cidr` was overwritten by `pod_cidrs[0]` (similar for `service_cidr`). Is this no longer needed?

By the way, I suppose it is a bit strange to provide both options `--pod-cidr` and `--pod-cidrs` at the same time, since option `--pod-cidrs` is sufficient to cover the function of option `--pod-cidr` (except for compatibility issues); maybe we can discuss with PM whether we need to discard option `--pod-cidr` in the future?

One suggestion to the current situation is, do we need to output a warning level log message (or even raise a `MutuallyExclusiveArgumentError`) when the user uses the two options `--pod-cidr` and `--pod-cidrs` at the same time, reminding the user that the value set by `--pod-cidr` will be overwritten by `--pod-cidrs` (or this is not allowed)?

Reply:
no need to overwrite the value as RP will take care of it if pod_cidr and service_cidr are omitted but the cidrs are supplied.